rclpy/rclpy/context.py

```diff
@@ -187,9 +187,11 @@ def get_domain_id(self) -> int:
             return self.__context.get_domain_id()
 
     def __enter__(self) -> 'Context':
-        # We do not accept parameters here. If one wants to customize the init() call,
-        # they would have to call it manually and not use the ContextManager convenience
-        self.init()
+        if self.__context is None:
+            # This object hasn't been initialized yet, so we can't use a context manager
+            raise RuntimeError(
+                'init() must be called on this Context before using it with a context manager')
```

Comment:
> It changes the context manager implementation of Context so that it does not call init() within itself. This definitely changes the semantics

i may be totally wrong...

this change is required because `managed_init` calls `init` including `context.init`? so if `managed_init` implements inline without calling `init` and `context.init`, this does not need to be changed?

Reply:
> this change is required because `managed_init` calls `init` including `context.init`? so if `managed_init` implements inline without calling `init` and `context.init`, this does not need to be changed?

Partially yes.  But I also think that the current implementation is just wrong.  Calling `init` inside of `__enter__`, without the ability to pass arguments, just seems incorrect to me.  I think it is much better to require that users already call `context.init()` before entering the context-manager.

Reply:
thanks for the explanation, yes that makes more sense to me too.